tests/middleware/test_auth_route.py

```diff
@@ -0,0 +1,174 @@
+        response = Response()
+        response.status_code = 401
+        response._content = b'{"message": "error"}'
+        return response
+
+    monkeypatch.setattr(requests, "post", mock_post)
+
+
+def test_simulated_existing_failed_auth(
+    client: TestClient, login_endpoint: str, dummy_user: UserModel, mock_post_failure: Any
+) -> None:
+    """Test the login endpoint with a failed authentication attempt
+
+    Arguments:
+        client (TestClient): The current test client.
+        login_endpoint (str): Endpoint prefix.
+        dummy_user (UserModel): A dummy user to test.
+        mock_post_failure (Any): this overwrites the request.post call
+    """
+    os.environ["AUTH_LOGIN_PROVIDER"] = "http://127.0.0.1"
+    os.environ["AUTH_TOKEN_ROUTE"] = login_endpoint
+    response = client.post(login_endpoint, json=dummy_user.dict())
+    assert response.status_code == 401
+
+
+def test_simulated_existing_auth(
+    client: TestClient, login_endpoint: str, dummy_user: UserModel, mock_post: Any
+) -> None:
+    """Test the login endpoint with a successful authentication attempt
+
+    Arguments:
+        client (TestClient): The current test client.
+        login_endpoint (str): Endpoint prefix.
+        dummy_user (UserModel): A dummy user to test.
+        mock_post (Any): this overwrites the request.post call
+    """
+    os.environ["AUTH_LOGIN_PROVIDER"] = "http://127.0.0.1"
+    os.environ["AUTH_TOKEN_ROUTE"] = login_endpoint
+    response = client.post(login_endpoint, json=dummy_user.dict())
+    assert response.status_code == 200
+
+
+def test_non_existing_auth(client: TestClient, login_endpoint: str, dummy_user: UserModel) -> None:
+    """Test the login endpoint with a non existing host and endpoint
+
+    Arguments:
+        client (TestClient): The current test client.
+        login_endpoint (str): Endpoint prefix.
+        dummy_user (UserModel): A dummy user to test.
+    """
+    os.environ["AUTH_LOGIN_PROVIDER"] = "http://127.0.0.1"
+    os.environ["AUTH_TOKEN_ROUTE"] = login_endpoint
+    response = client.post(login_endpoint, json=dummy_user.dict())
+    assert response.status_code == 401
+
+
+def test_refresh_forged(client: TestClient, refresh_endpoint: str) -> None:
+    """Test the refresh endpoint with a invalid token
+
+    Arguments:
+        client (TestClient): The current test client.
+        refresh_endpoint (str): Endpoint prefix.
+    """
+    response = client.post(refresh_endpoint, headers={"Authorization": "Bearer 123"})
+    assert response.status_code == 401
+
+
+def test_refresh(client: TestClient, refresh_endpoint: str, dummy_token: str) -> None:
+    """Test the refresh endpoint with a valid token
+
+    Arguments:
+        client (TestClient): The current test client.
+        refresh_endpoint (str): Endpoint prefix.
+        dummy_token (str): a valid dummy token
+    """
+    response = client.post(refresh_endpoint, headers={"Authorization": f"Bearer {dummy_token}"})
```

Comment:
Can we please validate that we retrieved a valid JWT token and that its expiration is longer than the dummy_token?